static/js/theme-map/VerticalFullPageMapOrchestrator.js

```diff
@@ -159,9 +165,87 @@ class VerticalFullPageMapOrchestrator extends ANSWERS.Component {
       this.searchThisArea();
     });
 
+    this.setupMapViewCssClasses();
     this.addMapComponent();
   }
 
+  /**
+   * Properly set CSS classes for mobile and desktop
+   */
+  setupMapViewCssClasses () {
+    if (!this.isMobile()) {
+      this.updateCssForDesktop();
+    }
+    window.addEventListener('resize', () => {
+      if (this.isMobile()) {
+        this.updateCssForMobile();
+      } else {
+        this.updateCssForDesktop();
+      }
+    }, { passive: true });
+  }
+
+  isMobile () {
+    return window.innerWidth <= this.mobileBreakpointMax;
+  }
+
+  updateCssForMobile () {
+    if (this._mobileView === MobileViews.LIST_VIEW) {
+      this.addMobileListViewCssClasses();
+      this.removeMobileMapViewCssClasses();
+      this.removeMobileDetailShownCssClasses();
+    } else if (this._mobileView === MobileViews.MAP_VIEW) {
+      this.addMobileMapViewCssClasses();
+      this.removeMobileListViewCssClasses();
+    }
+  }
+
+  updateCssForDesktop () {
+    this.removeMobileListViewCssClasses();
+    this.removeMobileMapViewCssClasses();
+    this.removeMobileDetailShownCssClasses();
+  }
+
+  setMobileMapView () {
+    this._mobileView = 'mapView';
+    this.updateCssForMobile();
+  }
+
+  setMobileListView () {
+    this._mobileView = 'listView';
+    this.updateCssForMobile();
+  }
+
+  addMobileListViewCssClasses () {
```

Comment:
How do you feel about some
```
applyCssClassesForState(state) {
    let classModifier = getModifierForState(state)
    this._container.classList.add('VerticalFullPageMap--${classModifier}');
    this._pageWrapperEl.classList.add('YxtPage-wrapper--${classModifier}');
}
```

where state -> css class modifier can come from your MobileViews.js enum?

Reply:
sure

Reply:
So I renamed MobileViews to MobileStates, so that we could include detailShown in there. So all three are states, however only map view and list view are views.